src/overlay005/field_control.c

```diff
@@ -0,0 +1,1092 @@
+{
+    Location nextMap;
+
+    if (Field_MapConnection(fieldSystem, playerX, playerZ, &nextMap) == FALSE) {
+        return FALSE;
+    }
+
+    if (sub_0205DC2C(curTileBehavior) == TRUE) {
+        int playerDir = PlayerAvatar_GetDir(fieldSystem->playerAvatar);
+
+        if (playerDir == DIR_WEST) {
+            playerDir = DIR_EAST;
+        } else if (playerDir == DIR_EAST) {
+            playerDir = DIR_WEST;
+        } else {
+            GF_ASSERT(FALSE);
+            return FALSE;
+        }
+
+        sub_02056BDC(fieldSystem, nextMap.unk_00, nextMap.unk_04, 0, 0, playerDir, 2);
+        return TRUE;
+    } else if (sub_0205DC38(curTileBehavior) == TRUE) {
+        int playerDir = PlayerAvatar_GetDir(fieldSystem->playerAvatar);
+
+        if (playerDir != DIR_WEST && playerDir != DIR_EAST) {
+            GF_ASSERT(FALSE);
+            return FALSE;
+        }
+
+        sub_02056BDC(fieldSystem, nextMap.unk_00, nextMap.unk_04, 0, 0, playerDir, 2);
+        return TRUE;
+    }
+
+    if (sub_0205DB10(curTileBehavior) || sub_0205DB40(curTileBehavior)) {
+        sub_02056C18(fieldSystem, nextMap.unk_00, nextMap.unk_04, 0, 0, 0);
+        return TRUE;
+    }
+
+    if (sub_0205DEE4(curTileBehavior)) {
+        sub_02053F58(fieldSystem, nextMap.unk_00, nextMap.unk_04);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static BOOL Field_UpdateDaycare (FieldSystem *fieldSystem)
+{
+    Party *party = Party_GetFromSavedata(fieldSystem->saveData);
+    UnkStruct_02026310 *v1 = sub_02026310(fieldSystem->saveData);
+
+    if (ov5_021E7154(v1, party, fieldSystem) == TRUE) {
+        UnkStruct_0202CD88 *v2 = sub_0202CD88(fieldSystem->saveData);
+
+        sub_0202CF28(v2, (1 + 10));
+        sub_0202CFEC(v2, 15);
+        sub_0203E880(fieldSystem, 2031, NULL);
+
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static BOOL Field_UpdateVsSeeker (FieldSystem *fieldSystem)
+{
+    ov5_021DBB94(fieldSystem);
+    return FALSE;
+}
+
+static BOOL Field_UpdatePokeRadar (FieldSystem *fieldSystem)
+{
+    RadarChargeStep(fieldSystem);
+    return FALSE;
+}
+
+static BOOL Field_UpdateRepel (FieldSystem *fieldSystem)
+{
+    return ov6_02246BF4(fieldSystem->saveData, fieldSystem);
+}
+
+static BOOL Field_UpdateFriendship (FieldSystem *fieldSystem)
+{
+    UnkStruct_020507E4 *events;
+    BOOL ret = FALSE;
+
+    events = SaveData_Events(fieldSystem->saveData);
+    u16 steps = sub_0206B44C(events);
+
+    steps++;
+
+    if (steps >= 128) {
+        steps = 0;
+        ret = TRUE;
+    }
+
+    sub_0206B45C(events, steps);
+
+    return ret;
+}
+
+static void Field_CalculateFriendship (FieldSystem *fieldSystem)
+{
+    int i, partyCount;
+    Pokemon *mon;
+    Party *party = Party_GetFromSavedata(fieldSystem->saveData);
+    u16 mapID = MapHeader_GetMapLabelTextID(fieldSystem->unk_1C->unk_00);
+
+    partyCount = Party_GetCurrentCount(party);
+
+    for (i = 0; i < partyCount; i++) {
+        mon = Party_GetPokemonBySlotIndex(party, i);
```

Comment:
```suggestion
    Party *party = Party_GetFromSavedata(fieldSystem->saveData);
    u16 mapID = MapHeader_GetMapLabelTextID(fieldSystem->unk_1C->unk_00);
    int partyCount = Party_GetCurrentCount(party);

    for (int i = 0; i < partyCount; i++) {
        Pokemon *mon = Party_GetPokemonBySlotIndex(party, i);
```

Reply:
I couldn't get this suggestion to match - the closest I could get is having `mon` declared as it's assigned a value. Couldn't get `i` or `partyCount` to match with C99-style.

Reply:
If you can't make it match with C99 style, I like to leave a small comment at the top that explains as such. It's meaningless information outside of matching, but comments are mostly irrelevant for dynamic documentation anyways.